src/NzbDrone.Core/Music/TrackService.cs

```diff
@@ -89,7 +89,7 @@ public Track FindTrackByTitle(int artistId, int albumId, int mediumNumber, strin
                     Length = Parser.Parser.NormalizeEpisodeTitle(track.Title).Length,
                     Track = track
                 })
-                                .Where(e => e.Track.Title.Length > 0 && e.Position >= 0)
+                                .Where(e => e.Track.Title.Length > 0 && e.Position >= 0 && e.Track.AbsoluteTrackNumber == trackNumber)
```

Comment:
I think there are some tracks that don't end up with a track length for some reason. Is there a reason we should ignore those? And just to make sure, `trackNumber` will be an absolute track number, correct?

Reply:
TrackNumber is absolute. As discussed, will tweak this logic a bit.